R/prune.R

```diff
@@ -0,0 +1,158 @@
+        paste(entries, collapse="','")))$id
+    if (any(!entries %in% db_entries)) {
+      entries <- paste(entries[!entries %in% db_entries],
+                       collapse = ", ")
+      stop(sprintf("%s not found: %s\n", table, entries))
+    }
+    db_entries
+  }
+
+  modelling_group <- thing_exists(e$prune_csv$modelling_group, "modelling_group")
+  disease <- thing_exists(e$prune_csv$disease, "disease")
+  touchstone <- thing_exists(e$prune_csv$scenario, "scenario_description")
+  scenario <- thing_exists(e$prune_csv$touchstone, "touchstone")
+
+
+  # For each row in the CSV file:
+
+  burden_estimate_set <- NULL
+
+  for (i in seq_len(nrow(e$prune_csv))) {
+    row <- e$prune_csv[i, ]
+    touchstone <- unlist(strsplit(row$touchstone, ";"))
+    modelling_group <- unlist(strsplit(row$modelling_group, ";"))
+    disease < unlist(strsplit(row$disease, ";"))
+    scenario <- unlist(strsplit(row$scenario, ";"))
+
+    touchstone <- if ("*" %in% touchstone) NULL else touchstone
+    modelling_group <- if ("*" %in% modelling_group) NULL else modelling_group
+    disease <- if ("*" %in% disease) NULL else disease
+    scenario <- if ("*" %in% scenario) NULL else scenario
+
+    # Find all the responsibilities for any of these fields, including
+    # the current_burden_estimate_set
+
+    resp_info <- DBI::dbGetQuery(con, sprintf("
+      SELECT modelling_group, disease,
+             scenario.touchstone as touchstone,
+             scenario.scenario_description as scenario,
+             responsibility.id as responsibility,
+             current_burden_estimate_set
+        FROM responsibility
+        JOIN responsibility_set
+          ON responsibility_set.id = responsibility.responsibility_set
+        JOIN scenario
+          ON responsibility.scenario = scenario.id
+        JOIN scenario_description
+          ON scenario_description.id = scenario.scenario_description
+          %s", get_filters(touchstone, modelling_group, disease, scenario)))
+
+    # Ignore where no burden_estimate_set has been uploaded.
+
+    resp_info <- resp_info[!is.na(resp_info$current_burden_estimate_set), ]
+
+    #################################################################
+    # Find all the burden estimate sets for the above
+    # responsibilities, that aren't the current_burden_estimate_set
+
+    burden_estimate_set <- rbind(burden_estimate_set,
+      DBI::dbGetQuery(con, sprintf("
+        SELECT *
+          FROM burden_estimate_set
+         WHERE responsibility IN %s
+           AND NOT id IN %s",
+        sql_in(c(-1, resp_info$responsibility)),
+        sql_in(c(-1, resp_info$current_burden_estimate_set)))))
+  }
+
+  # We now have a list of burden estimate sets to cull -
+  # Remove any duplicates, and return
+
+  burden_estimate_set <- burden_estimate_set[!duplicated(burden_estimate_set$id), ]
+  list(burden_estimate_set = burden_estimate_set)
+}
+
+test_extract_prune <- function(e) {
+
+}
```

Comment:
Missing some impl?

Reply:
Yeah... there's not a lot left to test here as the cases that would fail in the build-in test_extract_prune function should already have stopped with a particular error message for the user, so I think these can be stubs. But it doesn't look very nice, so I'll look at this again...

Reply:
Added a couple of tests here just to see that if burden_estimate_set is present, it has the fields we're interested in